src/04-spin-boson-model.md

```diff
@@ -0,0 +1,800 @@
+
+
+
+**Important detour on state numbers and labels**
+
+We'll study Fig 4 in more detail shortly, but for now I want to draw you attention to the labels for the rows and columns. For example:
+- $|3, 0 \rangle$ represents 3 bosons and a TSS state of |+>
+- $|3, 1 \rangle$ represents 3 bosons and a TSS state of |->
+
+It is hard to remember QuTiP's convention for the numbering of states. It is therefore very helpful to have a way to map these QuTiP number states to something more immediately recognisable - let's create such a map.
+
+```python
+possible_ns = range(0, max_bosons+1)
+possible_ms = ["+","-"]
+nm_list = [(n,m) for (n,m) in product(possible_ns, possible_ms)]
+```
+
+```python
+nm_list
+```
+
+Whenever we want to know what |n,±> state corresponds to QuTiP's number state e.g. 4, we can just do this:
+
+```python
+nm_list[4]
+```
+
+We can create some nice labels corresponding to the `nm_list`. This will make things plots easier to understand.
+
+```python
+def make_braket_labels(nm_list):
+    bra_labels = ["$\langle$"+str(n)+", "+str(m)+" |" for (n,m) in nm_list]
+    ket_labels = ["| "+str(n)+", "+str(m)+"$\\rangle$" for (n,m) in nm_list]
+    return bra_labels, ket_labels
+```
+
+```python
+bra_labels, ket_labels = make_braket_labels(nm_list)
+```
+
+We can now pass these labels into the Hinton diagram
+
+```python
+f, ax = hinton(H, xlabels=ket_labels, ylabels=bra_labels)
+ax.tick_params(axis='x',labelrotation=90,)
+ax.set_title("Matrix elements of H     (Fig 5)");
+```
+
+That's better!
+
+If we now take a closer look at the structure of the Hinton diagram we can see some interesting features when we follow a path that connects one state to another:
+
+```python
+print("                Matrix elements of H     (Fig 6)")
+Image(filename='./img/04-hinton.png') 
+```
+
+If we imagine starting a simulation with 0 bosons and the TSS in its + state, i.e. |0,+>, then Fig 6 suggests that:
+1. there are connections (albeit indirect) from |0,+> to many different states with many more bosons, e.g. $|0,+> \rightarrow |1,-> \rightarrow |2,+> \rightarrow |3,-> \rightarrow |4,+> ...$
+2. there are some states that are not accessible at all if we start in the |0,+> state
+
+
+On 1. These indirect connections provide a mechanism to achieve the potential down conversion that we saw hints of earlier (namely $|0,+> \rightarrow |3,->$).
+
+On 2. The Hamiltonian appears to be composed of two separate "universes" that don't interact with each other. In our energy level diagram (Fig 2) both universes are present - perhaps if we separate them we'll only see anti-crossings in the respective plots.
+
+We're getting closer to convincing ourselves of the reality of down conversion, but before we check this through simulation we need to figure out how to separate the two universes.
```

Comment:
I'm struggling a bit here and it might just be because I don't have some of the basics down to the extent that I should. Two things come to mind where I would benefit from some more comments. 
1) Maybe an extra sentence or two "how we can move" across the Hinton diagram and how not (and why). 
2) Maybe a comment about energy conversion. If we start at |0,+> and end at |4,+> where did the energy for the 4 bosons come from?

Reply:
yes
1. the idea of moving  is something like: you are in a certain state x, and being in a certain state means that you are in row x: in that row there are as many values as the columns are, i.e. all the other states: Travelling in that matrix simply means that the color of the pixel represent the probability of you in state x "transforming" in state Y. and so on ...

2. |0,+> was not an eigenstate of the energy, indeed this state oscillates. The additional energy is provided by YOU measuring the energy

Reply:
Great points, Nicola. I found those comments very helpful, thanks!